mediatek: filogic: add support for Zbtlink ZBT-Z8802BE

ysc3839 opened the pull request:
Zbtlink ZBT-Z8802BE is a Wi-Fi 7 BE7200 router with (optional) 5G WWAN support.
OEM web page: https://www.zbt-china.com/productinfo/3636055.html

### Hardware
MediaTek MT7988D SoC (3x Cortex-A73 @1.8 GHz max)
512MB DDR4 RAM
128MB SPI-NAND
MediaTek MT7992AV+MT7975N+MT7979N BE7200 Dual-Band Wi-Fi 7
4x LAN (4x 1GE MT7988 built-in)
1x WAN (2.5GE MT7988 built-in)
LED: 2x 5G, 1x RGB SYS
USB: 1x USB 3
Buttons: RESET, MESH
UART: 115200 8N1 3.3V

### Installation
1. Hold down RESET button and power on the device

2. Assign IP 192.168.1.2/24 to your computer's Ethernet port

3. Connect Ethernet to one of the 1GE LAN ports

4. Open browser and visit http://192.168.1.1

5. Upload zbtlink_zbt-z8802be-squashfs-sysupgrade.bin

The one I have does not have 5G WWAN module, so I can't test if it works.
OEM provided files (dts, GPIO definition): [zbt-z8802be.zip](https://github.com/user-attachments/files/21673961/zbt-z8802be.zip)

rbqvq commented:
Also add `gpio_switch`
You can find a example at `https://github.com/openwrt/openwrt/blob/main/target/linux/ramips/mt76x8/base-files/etc/board.d/03_gpio_switches`

ysc3839 commented:
Diff after rebase:
```diff
diff --git a/target/linux/mediatek/dts/mt7988d-zbtlink-zbt-z8802be.dts b/target/linux/mediatek/dts/mt7988d-zbtlink-zbt-z8802be.dts
index 18bf619bf9..875a0abff8 100644
--- a/target/linux/mediatek/dts/mt7988d-zbtlink-zbt-z8802be.dts
+++ b/target/linux/mediatek/dts/mt7988d-zbtlink-zbt-z8802be.dts
@@ -23,15 +23,16 @@
 	};
 
 	chosen {
-		bootargs = "console=ttyS0,115200n1 loglevel=8 earlycon=uart8250,mmio32,0x11000000 pci=pcie_bus_perf";
+		bootargs = "console=ttyS0,115200n1 earlycon=uart8250,mmio32,0x11000000 pci=pcie_bus_perf";
 	};
 
 	cpus {
 		/delete-node/ cpu@3;
 	};
 
-	memory {
+	memory@40000000 {
 		reg = <0x0 0x40000000 0x0 0x10000000>;
+		device_type = "memory";
 	};
 
 	gpio-export {
@@ -48,7 +49,7 @@
 				gpio-export,output = <0>;
 				gpios = <&pio 61 GPIO_ACTIVE_HIGH>;
 		};
-    };
+	};
 
 	gpio-keys {
 		compatible = "gpio-keys";
```

ysc3839 committed:
mediatek: filogic: add support for Zbtlink ZBT-Z8802BE

Hardware
--------
MediaTek MT7988D SoC (3x Cortex-A73 @1.8 GHz max)
512MB DDR4 RAM
128MB SPI-NAND
MediaTek MT7992AV+MT7975N+MT7979N BE7200 Dual-Band Wi-Fi 7
4x LAN (4x 1GE MT7988 built-in)
1x WAN (2.5GE MT7988 built-in)
LED: 2x 5G, 1x RGB SYS
USB: 1x USB 3
Buttons: RESET, MESH
UART: 115200 8N1 3.3V

Installation
------------
1. Hold down RESET button and power on the device

2. Assign IP 192.168.1.2/24 to your computer's Ethernet port

3. Connect Ethernet to one of the 1GE LAN ports

4. Open browser and visit http://192.168.1.1

5. Upload zbtlink_zbt-z8802be-squashfs-sysupgrade.bin

Signed-off-by: Richard Yu <yurichard3839@gmail.com>

ysc3839 commented:
@DragonBluep Updated. Thanks.

DragonBluep commented on target/linux/mediatek/dts/mt7988d-zbtlink-zbt-z8802be.dts:
```diff
@@ -0,0 +1,409 @@
+		#address-cells = <3>;
+		#size-cells = <2>;
+
+		wifi@0,0 {
```
Please add `compatible = "mediatek,mt76";`.

DragonBluep commented on target/linux/mediatek/dts/mt7988d-zbtlink-zbt-z8802be.dts:
```diff
@@ -0,0 +1,409 @@
+			band@1 {
+				reg = <1>;
+				nvmem-cells = <&macaddr_factory_a>;
+				nvmem-cell-names = "mac-address";
```
Can you rebase and check if it is working now? If not, please set the MAC in script `11_fix_wifi_mac`.

joelinux60 replied:
Would like to share that I got mac address assignment working after adding index numbers to nvmem-cells like below:

&wifi {
	status = "okay";
	nvmem-cells = <&eeprom_factory>;
	nvmem-cell-names = "eeprom";

	band@0 {
		reg = <0>;
		nvmem-cells = <&macaddr_factory_4 0>;
		nvmem-cell-names = "mac-address";
	};

	band@1 {
		reg = <1>;
		nvmem-cells = <&macaddr_factory_a 1>;
		nvmem-cell-names = "mac-address";
	};
};

Did it in my #PR #19823

DragonBluep commented on target/linux/mediatek/dts/mt7988d-zbtlink-zbt-z8802be.dts:
```diff
@@ -0,0 +1,409 @@
+		compatible = "spi-nand";
+		reg = <0>;
+		#address-cells = <1>;
+		#size-cells = <1>;
```
Remove `#address-cells` and `#size-cells`.
